model/task/task_test.go

```diff
@@ -3669,6 +3669,156 @@ func TestArchive(t *testing.T) {
+		t2 := Task{
+			Id:      "t2",
+			Status:  evergreen.TaskSucceeded,
+			Version: "v",
+		}
+		assert.NoError(t, t2.Insert())
+		dt := Task{
+			Id:                      "dt",
+			DisplayOnly:             true,
+			ExecutionTasks:          []string{"t1", "t2"},
+			Status:                  evergreen.TaskFailed,
+			Version:                 "v",
+			ResetFailedWhenFinished: true,
+		}
+		assert.NoError(t, dt.Insert())
+
+		return dt
+	}
+
+	checkTaskIsArchived := func(t *testing.T, oldTaskID string) {
+		dbTask, err := FindOneOldId(oldTaskID)
+		require.NoError(t, err)
+		require.NotZero(t, dbTask)
+		assert.NotZero(t, dbTask.OldTaskId)
+		assert.NotEqual(t, dbTask.OldTaskId, dbTask.Id)
+		assert.True(t, dbTask.Archived)
+		assert.False(t, dbTask.Aborted)
+		assert.Zero(t, dbTask.AbortInfo)
+	}
+
+	checkTaskIsNotArchived := func(t *testing.T, taskID string, execution int) {
+		task, err := FindOneIdAndExecution(taskID, execution)
+		assert.NoError(t, err)
+		assert.False(t, task.Archived)
+
+		oldT, err := FindOneOldId(MakeOldID(taskID, execution))
+		assert.NoError(t, err)
+		assert.Nil(t, oldT)
+
+		nextExecution, err := FindOneIdAndExecution(taskID, execution+1)
+		assert.NoError(t, err)
+		assert.Nil(t, nextExecution)
+	}
+
+	checkEventLogHostTaskExecutions := func(t *testing.T, hostID, oldTaskID string, execution int) {
+		dbTask, err := FindOneOldId(oldTaskID)
+		require.NoError(t, err)
+		require.NotZero(t, dbTask)
+
+		events, err := event.FindAllByResourceID(hostID)
+		require.NoError(t, err)
+		assert.NotEmpty(t, events)
+
+		for _, e := range events {
+			hostEventData, ok := e.Data.(*event.HostEventData)
+			require.True(t, ok)
+			require.Equal(t, hostEventData.TaskId, dbTask.OldTaskId)
+			require.Equal(t, hostEventData.Execution, strconv.Itoa(dbTask.Execution), len(events))
+		}
+	}
+
+	dt := resetDatabase()
+
+	t.Run("ArchivesOnlyFailedExecutionTasks", func(t *testing.T) {
+		require.True(t, dt.ResetFailedWhenFinished)
+		t1, err := FindOneIdAndExecution(dt.ExecutionTasks[0], dt.Execution)
+		require.NoError(t, err)
+		t2, err := FindOneIdAndExecution(dt.ExecutionTasks[1], dt.Execution)
+		require.NoError(t, err)
+
+		archivedT1 := MakeOldID(t1.Id, t1.Execution)
+		archivedExecution := t1.Execution
+
+		hostID := "hostID"
+		event.LogHostRunningTaskSet(hostID, t1.Id, 0)
+		event.LogHostRunningTaskCleared(hostID, t1.Id, 0)
+
+		archivedDisplayTaskID := MakeOldID(dt.Id, dt.Execution)
+		require.Equal(t, 0, dt.Execution)
+		require.NoError(t, dt.Archive())
+		dtPointer, err := FindOneId(dt.Id)
+		dt = *dtPointer
+		require.NoError(t, err)
+		require.Equal(t, 1, dt.Execution)
+
+		checkTaskIsArchived(t, archivedT1)
+		checkTaskIsNotArchived(t, t2.Id, 0)
+		checkTaskIsArchived(t, archivedDisplayTaskID)
+
+		checkEventLogHostTaskExecutions(t, hostID, archivedT1, archivedExecution)
+	})
+
+	// This test is for the edge case of a archiving with only failed execution tasks, then archiving all execution tasks
```

Comment:
can you test this behavior with other tasks in here as well as the display task?

Reply:
Yes, let me add some more tasks!

Reply:
(to clarify, I'll make a new test for this because this is specifically testing calling archive on a display task

Reply:
im not sure where the newly added test is that looks at t3 and t4